bitnami/mariadb/templates/primary/statefulset.yaml

```
@@ -318,14 +318,16 @@ spec:
           {{- if .Values.metrics.resources }}
           resources: {{- toYaml .Values.metrics.resources | nindent 12 }}
           {{- end }}
-          {{- if and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles) }}
+          {{- if or (and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles)) .Values.metrics.extraVolumeMounts.primary }}
           volumeMounts:
+          {{- if and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles) }}
             - name: mariadb-credentials
               mountPath: /opt/bitnami/mysqld-exporter/secrets/
           {{- end }}
           {{- if .Values.metrics.extraVolumeMounts.primary }}
           {{- include "common.tplvalues.render" (dict "value" .Values.metrics.extraVolumeMounts.primary "context" $) | nindent 12 }}
           {{- end }}
+          {{- end }}
```

The extra conditionals are not really needed, could you please refactor this as it follows?

```suggestion
          volumeMounts:
          {{- if and .Values.auth.usePasswordFiles (not .Values.auth.customPasswordFiles) }}
            - name: mariadb-credentials
              mountPath: /opt/bitnami/mysqld-exporter/secrets/
          {{- end }}
          {{- if .Values.metrics.extraVolumeMounts.primary }}
          {{- include "common.tplvalues.render" (dict "value" .Values.metrics.extraVolumeMounts.primary "context" $) | nindent 12 }}
          {{- end }}
```

---

I added the conditionals as I was not sure how Kubernetes would react, if the key `volumeMounts` is rendered, but without any list entries (not even an empty list). This would happen if neither `.Values.auth.usePasswordFiles` nor `.Values.metrics.extraVolumeMounts` are used. But I trust your expertise and I will follow your suggestion.

---

Hi @tstollin, no worries. 
Kubernetes won't fail if an empty `volumeMounts` is rendered, the same approach is used in other charts, e.g. airflow:
https://github.com/bitnami/charts/blob/21cc5f9ab8127da23f271f085a2e8bd916c2cc0f/bitnami/airflow/templates/worker/statefulset.yaml#L182-L202

---

Perfect. Thanks for the clarification.